store/src/java/com/zimbra/cs/service/util/VolumeConfigUtil.java

```diff
@@ -99,8 +100,16 @@ public static void validateCreateVolumeRequest(CreateVolumeRequest request, Volu
             true != volInfoRequest.isCurrent()) {
             throw VolumeServiceException.INVALID_REQUEST("Volume Current can be TRUE, FALSE or OPTIONAL(don't provide)", VolumeServiceException.BAD_VOLUME_CURRENT);
         }
+        
+        // validate storage type
+        String storageTypeS3 = volInfoRequest.getVolumeExternalInfo().getStorageType();
```

Comment:
Add checks here, whether it is S3 volume or OpenIO volume.. you don't need to accept both. Even we should restrict one only.

Reply:
added condition inside external